inlang/source-code/sdk/src/persistence/filelock/acquireFileLock.ts

```diff
@@ -21,16 +21,17 @@ export async function acquireFileLock(
 	try {
 		debug(lockOrigin + " tries to acquire a lockfile Retry Nr.: " + tryCount)
 		await fs.mkdir(lockDirPath)
-		const stats = await fs.stat(lockDirPath)
+		const stats = await fs.stat(lockDirPath) // not atomic with mkdir, but good enough
```

Comment:
```suggestion
		// NOTE: fs.stat does not need to be atomic since mkdir would crash atomically - if we are here its save to consider the lock held by this process
		const stats = await fs.stat(lockDirPath) 
```
I agree a comment on this is needed to clearify the reasoning and i added a suggestion. e should understand why this is good enough - if the comment doesn't help let me know so we can be more precise